R/SupervisedTask.R

```diff
@@ -1,13 +1,13 @@
-makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data) {
-  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data)
+makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data, spatial) {
+  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data, spatial = spatial)
 
   if (check.data) {
     # costsens does not have a target col...
     # assertCharacter(target, any.missing = FALSE, min.len = 1L)
     w = which.first(target %nin% colnames(data))
     if (length(w) > 0L)
       stopf("Column names of data doesn't contain target var: %s", target[w])
-    checkTaskData(task$env$data, cols = setdiff(colnames(data), target))
+    checkTaskData(task$env$data, cols = setdiff(colnames(task$env$data), target))
```

Comment:
See my question above -- does building multiple models with the same task work? It seems to me that as soon as you're removing `x` and `y` this will break.